### ISSUE TITLE
Relaxing GCC pins (currently using GCC 7)

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/cudadecon-feedstock#19]
It looks like GCC is currently pinned in this recipe (and has been for a while). Wondering if it is possible to relax this pinning and use the conda-forge default GCC (currently [GCC 12]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/cbaccc9bd82d0fc4c3ae331245d2b77759ce863a/recipe/conda_build_config.yaml#L6-L7 ))

https://github.com/conda-forge/cudadecon-feedstock/blob/df80cf2203a8b69937bd65b8d4c002dab2cd794a/recipe/conda_build_config.yaml#L1-L6

**Comment by tlambert03** [comment]
Hey @jakirkham, thanks for the note and sorry about the long pinning.  I'm absolutely open to relaxing anything.  I just need to find time to test stuff (and really, if it builds successfully, that's good enough for me I think).   Is this the same thing as https://github.com/conda-forge/cudadecon-feedstock/pull/18?  (i.e. is it still waiting on me/us to update the underlying library), or could it be as simple as just changing the config?

**Comment by jakirkham** [comment]
Not having much context on the issues these pinnings were trying to solve (they appear to have been part of this feedstock since the beginning), can't really say what issues we would encounter

In theory this should be as simple as dropping the `conda_build_config.yaml` file and re-rendering. Though Idk what might come up after that. Things could be fine or we could re-encounter some form of the original issue that motivated the pinnings

We could at least try and see

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by tlambert03** [comment]
> Not having much context on the issues these pinnings were trying to solve (they appear to have been part of this feedstock since the beginning), can't really say what issues we would encounter

@zbarry originally (graciously) set up this feedstock.  and I recall a little back and forth on the conda-forge gitter, but I don't recall the  details.

let's try!

**Comment by jakirkham** [comment]
Gotcha maybe he has more insight 🙂

Though imagine lots has changed in the ~2yrs since this feedstock was setup

Yeah the main thing is the `conda_build_config.yaml` is causing the feedstock issues re-rendering ( for example: https://github.com/conda-forge/cudadecon-feedstock/pull/20#issuecomment-1690778040 ), which may be causing migrations and version updates to get lost (since the bot will fail to open those due to re-rendering issues). So at least resolving the re-rendering issues (even if some kind of pin is still needed) would be an improvement. Though ideally no pin will be needed (making things more compatible with conda-forge on the whole). Hopefully that provides a bit more context on the issue

**Comment by jakirkham** [comment]
Ok have dropped the `conda_build_config.yaml` from the bot's PR, re-rendered, and bumped the `build/number` (since a new GCC will be used on Linux)

If it passes, we should be good to go

If it fails, we will at least have better insight into any remaining issues

**Comment by tlambert03** [comment]
thank you much!

**Comment by zbarry** [comment]
Thanks both! Don't remember precisely the reason for the pinning, but glad it seems to be working.

**Comment by jakirkham** [comment]
Looks like it was due to an issue building with old CUDA versions that had upper bounds on the GCC use ( https://github.com/conda-forge/staged-recipes/pull/14423#issuecomment-812157781 )

Apparently I didn't remember this either ( https://github.com/conda-forge/staged-recipes/pull/14423#issuecomment-812165914 ) 🤦‍♂️ 

Anyways this is no longer needed as we tie the compiler versions and CUDA versions together in global pinnings. So there is no need to set this sort of thing per recipe (unless there are more specific compilation constraints that the recipe has)

**Comment by tlambert03** [comment]
😂
